lib/workers/repository/process/vulnerabilities.ts

```diff
@@ -91,51 +97,392 @@ export class Vulnerabilities {
       { manager, packageFile, queueLength: queue.length },
-      'fetchManagerPackagerFileUpdates starting with concurrency'
+      'fetchManagerPackageFileVulnerabilities starting with concurrency'
     );
 
     config.packageRules?.push(...(await p.all(queue)).flat());
-    logger.trace({ packageFile }, 'fetchManagerPackagerFileUpdates finished');
+    logger.trace(
+      { packageFile },
+      'fetchManagerPackageFileVulnerabilities finished'
+    );
   }
 
   private async fetchDependencyVulnerabilities(
     packageFileConfig: RenovateConfig & PackageFile,
-    packageDependency: PackageDependency
+    dep: PackageDependency
   ): Promise<PackageRule[]> {
-    const ecosystem =
-      Vulnerabilities.managerEcosystemMap[packageFileConfig.manager!];
+    const ecosystem = Vulnerabilities.datasourceEcosystemMap[dep.datasource!];
+    if (!ecosystem) {
+      logger.trace(`Cannot map datasource ${dep.datasource!} to OSV ecosystem`);
+      return [];
+    }
+
+    let packageName = dep.packageName ?? dep.depName!;
+    if (ecosystem === 'PyPI') {
+      // https://peps.python.org/pep-0503/#normalized-names
+      packageName = packageName.toLowerCase().replace(regEx(/[_.-]+/g), '-');
+    }
+
+    const packageRules: PackageRule[] = [];
+    try {
+      const vulnerabilities = await this.osvOffline?.getVulnerabilities(
+        ecosystem,
+        packageName
+      );
+      if (
+        is.nullOrUndefined(vulnerabilities) ||
+        is.emptyArray(vulnerabilities)
+      ) {
+        logger.trace(
+          `No vulnerabilities found in OSV database for ${packageName}`
+        );
+        return [];
+      }
+
+      const depVersion =
+        dep.lockedVersion ?? dep.currentVersion ?? dep.currentValue!;
+
+      const versioning = dep.versioning ?? getDefaultVersioning(dep.datasource);
+      const versioningApi = getVersioning(versioning);
+
+      if (!versioningApi.isVersion(depVersion)) {
+        logger.debug(
+          `Skipping vulnerability lookup for package ${packageName} due to unsupported version ${depVersion}`
+        );
+        return [];
+      }
+
+      for (const vulnerability of vulnerabilities) {
+        for (const affected of vulnerability.affected ?? []) {
+          const isVulnerable = this.isPackageVulnerable(
+            ecosystem,
+            packageName,
+            depVersion,
+            affected,
+            versioningApi
+          );
+          if (!isVulnerable) {
+            continue;
+          }
+
+          logger.debug(
+            `Vulnerability ${vulnerability.id} affects ${packageName} ${depVersion}`
+          );
+          const fixedVersion = this.getFixedVersion(
+            ecosystem,
+            depVersion,
+            affected,
+            versioningApi
+          );
+          if (is.nullOrUndefined(fixedVersion)) {
+            logger.debug(
+              `No fixed version available for vulnerability ${vulnerability.id} in ${packageName} ${depVersion}`
```

Comment:
Makes sense for me. I did a check on a dump of all supported vuln dbs to see how noisy this could be:
there are currently ~51.000 events, of which ~750 don't specify a `fixed` or `last_affected` version -> 1.5%